### PR TITLE
expression,types: improve `str_to_date` builtin function compatibility

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

mysql:
```
mysql> select str_to_date('10/28/2011 9:46:29 Am','%m/%d/%Y %l:%i:%s %p');
+-------------------------------------------------------------+
| str_to_date('10/28/2011 9:46:29 Am','%m/%d/%Y %l:%i:%s %p') |
+-------------------------------------------------------------+
| 2011-10-28 09:46:29                                         |
+-------------------------------------------------------------+
1 row in set (0.00 sec)
```

tidb:
```
mysql> select str_to_date('10/28/2011 9:46:29 Am','%m/%d/%Y %l:%i:%s %p');
+-------------------------------------------------------------+
| str_to_date('10/28/2011 9:46:29 Am','%m/%d/%Y %l:%i:%s %p') |
+-------------------------------------------------------------+
| NULL                                                        |
+-------------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)
```

%p formator should accept both AM/PM and am/pm, it's case insensitive

### What is changed and how it works?

Tiny fix

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @winkyao @lysu

**Comment by shenli** [comment]
LGTM
Please add labels for this PR.

File: expression/builtin_time_test.go [hunk 1]
```diff
@@ -1257,6 +1257,7 @@ func (s *testEvaluatorSuite) TestStrToDate(c *C) {
 		Success bool
 		Expect  time.Time
 	}{
+		{"10/28/2011 9:46:29 pm", "%m/%d/%Y %l:%i:%s %p", true, time.Date(2011, 10, 28, 21, 46, 29, 0, time.Local)},
```
Comment (shenli) [inline expression/builtin_time_test.go]: Could you add a test case with "am"?

Comment (tiancaiamao) [reply expression/builtin_time_test.go]: Done

**Pull request merged**